testsuite/test_ta_workflow.py

```diff
@@ -0,0 +1,776 @@
+    '''
+        Check if users can not recompute score for the submissions
+        that are from a course where they are not teaching assistants
+    '''
+
+    (_, ta, user, _, _) = base_db_setup
+
+    course = Course.objects.create(name='PC')
+    assignment = course.assignment_set.create(
+        name='a0',
+        max_score=100,
+        deadline_soft=datetime(2050, 1, 1, tzinfo=timezone.utc),
+        deadline_hard=datetime(2050, 1, 1, tzinfo=timezone.utc),
+        repo_url='https://github.com/vmck/assignment',
+        repo_path='pc-00',
+    )
+
+    submission = assignment.submission_set.create(
+        score=100.00,
+        state=Submission.STATE_DONE,
+        user=user,
+    )
+
+    client.login(username=ta.username, password='pw')
+
+    response = client.post(
+        f'/submission/{submission.pk}/recompute',
+        follow=True
+    )
+
+    assert response.status_code == 403
+
+
+@pytest.mark.django_db
+def test_ta_download_last_sub(client, base_db_setup, mock_evaluate):
+    '''
+        Check if users can download the last submission for review
+        if they are teaching assistants
+    '''
+
+    (_, ta, _, _, _) = base_db_setup
+
+    client.login(username=ta.username, password='pw')
+
+    response = client.post(
+        f'/admin/interface/assignment/',
+        data={
+            'action': 'download_review_submissions',
+            '_selected_action': '1'
+        },
+        follow=True
+    )
+
+    assert response.status_code == 200
+    assert response.json()['type'] == "download_review_submissions"
+
+
+@pytest.mark.django_db
+def test_ta_download_all_subs(client, base_db_setup, mock_evaluate):
+    '''
+        Check if users can download all the submissions for review
+        if they are teaching assistants
+    '''
+
+    (_, ta, _, _, _) = base_db_setup
+
+    client.login(username=ta.username, password='pw')
+
+    response = client.post(
+        f'/admin/interface/assignment/',
+        data={
+            'action': 'download_all_submissions',
+            '_selected_action': '1'
+        },
+        follow=True
+    )
+
+    assert response.status_code == 200
+    assert response.json()['type'] == "download_all_submissions"
+
+
+@pytest.mark.django_db
+def test_ta_run_moss(client, base_db_setup, mock_evaluate):
+    '''
+        Check if users can run moss if they are teaching assistants
+    '''
+
+    (_, ta, user, _, _) = base_db_setup
+
+    client.login(username=ta.username, password='pw')
+
+    response = client.post(
+        f'/admin/interface/assignment/',
+        data={
+            'action': 'run_moss',
+            '_selected_action': '1'
+        },
+        follow=True
+    )
+
+    assert response.status_code == 200
+    assert response.json()['type'] == "run_moss"
+
+
+@pytest.mark.django_db
+def test_ta_login(client, STC, base_db_setup):
+    (_, ta, _, _, _) = base_db_setup
+    client.post(
+        '/',
+        {'username': ta.username, 'password': 'pw'}
```

Comment:
Final comma

Reply:
Is there a comma required here if I put it on the same line?